# Hypotéza 1
### Na silnicích první třídy se při nehodách umíralo se stejnou pravděpodobností jako dálnicích.
načteme knihovny pandas a scipy

In [1]:
import pandas as pd
import scipy.stats as st

Načteme Dataframe z accidents.pkl.gz

In [2]:
df = pd.read_pickle("accidents.pkl.gz")
df

p1  p36    p37         p2a  weekday(p2a)   p2b  p6  p7  p8  \
0       002100160001    4     -1  2016-01-01             5    55   1   1   0   
1       002100160002    4     -1  2016-01-01             5   130   1   3   0   
2       002100160003    5     -1  2016-01-01             5   100   1   2   0   
3       002100160004    6     -1  2016-01-01             5   120   9   0   0   
4       002100160005    6     -1  2016-01-01             5  2560   2   0   0   
...              ...  ...    ...         ...           ...   ...  ..  ..  ..   
572929  190906210457    6     -1  2021-08-30             1   506   2   0   0   
572930  190906210458    6     -1  2021-08-30             1   752   2   0   0   
572931  190906210459    3   2099  2021-08-30             1  1505   1   3   0   
572932  190906210460    3   2099  2021-08-31             2  1245   0   0   0   
572933  190906210461    3  21029  2021-08-31             2  1834   3   0   4   

        p9  ...      l        n  o                       p       q       r  \
0        2  ...          711403     Souhlasnýsesměremúseku  Pomalý  554782   
1        2  ...                                                         -1   
2        2  ...                                                         -1   
3        2  ...                                                         -1   
4        2  ...                                                         -1   
...     ..  ...    ...      ... ..                     ...     ...     ...   
572929   2  ...         2611387                             Pomalý  560359   
572930   2  ...         2541503         Opačnýkesměruúseku  Pomalý  560383   
572931   2  ...                                                         -1   
572932   1  ...   2099   199764         Opačnýkesměruúseku  Pomalý  560286   
572933   1  ...  21029  3650728     Souhlasnýsesměremúseku  Pomalý  560324   

             s                   t  p5a  region  
0       451622  GN_V0.1UIR-ADR_410    1     PHA  
1           -1                        1     PHA  
2           -1                        1     PHA  
3           -1                        1     PHA  
4           -1                        1     PHA  
...        ...                 ...  ...     ...  
572929      -1  GN_V0.1UIR-ADR_410    1     KVK  
572930  550426  GN_V0.1UIR-ADR_410    1     KVK  
572931      -1                        1     KVK  
572932  553654  GN_V0.1UIR-ADR_410    1     KVK  
572933      -1  GN_V0.1UIR-ADR_410    2     KVK  

[572934 rows x 65 columns]

Vytvoříme nové sloupce pro úmrtí na silnici, pro silnici první třídy a dálnici.
Vyloučíme možnost neúmrtí na silnici.
uložíme nehody na silnicích první třídy
uložíme nehody na dálnicích

In [3]:
df['umrti'] = df['p13a'] > 0
df['prvnitrida'] = df['p36'] == 1
df['dalnice'] = df['p36'] == 0

Vytvoříme si kontingenční tabulky pomocí funkce crosstab z knihovny pandas.

In [4]:
ct = pd.crosstab(df['umrti'], df['prvnitrida'])
dt = pd.crosstab(df['umrti'], df['dalnice'])
ct

prvnitrida   False   True
umrti                    
False       491637  78618
True          1768    911

In [5]:
dt

dalnice   False   True
umrti                 
False    545962  24293
True       2513    166

Pro tyto tabulky vypočítáme chi test pomocí funkce chi2_contingency z knihovny scipy. Tato funkce spočítá  statistiku a p-hodnotu pro test nezávislosti v kontingenční tabulce. Tato funkce vrací více hodnot, nás zajímá hodnoty expected array

In [6]:
prvnitrida = st.chi2_contingency(ct)
prvnitrida

(910.1658459006459,
 6.052038642793102e-200,
 1,
 array([[4.91097872e+05, 7.91571279e+04],
        [2.30712786e+03, 3.71872137e+02]]))

In [7]:
dalnice = st.chi2_contingency(dt)
dalnice

(23.99118835576904,
 9.67776014503534e-07,
 1,
 array([[5.45910369e+05, 2.43446314e+04],
        [2.56463140e+03, 1.14368603e+02]]))

In [8]:
st.stats.ttest_ind(dalnice[3],prvnitrida[3])

Ttest_indResult(statistic=array([ 0.07535067, -0.66810574]), pvalue=array([0.9467945 , 0.57284593]))

## Závěr hypotézy 1
pvalue nám ukazuje, že hodnoty nejsou v vybraném prahu spolehlivosti 5%
a proto hypotézu zavrhujeme

# Hypotéza 2
### Při nehodách vozidel značky Škoda je škoda na vozidle nižší než při nehodách vozidel Audi.
načteme Dataframe

In [9]:
df = pd.read_pickle("accidents.pkl.gz")
df

p1  p36    p37         p2a  weekday(p2a)   p2b  p6  p7  p8  \
0       002100160001    4     -1  2016-01-01             5    55   1   1   0   
1       002100160002    4     -1  2016-01-01             5   130   1   3   0   
2       002100160003    5     -1  2016-01-01             5   100   1   2   0   
3       002100160004    6     -1  2016-01-01             5   120   9   0   0   
4       002100160005    6     -1  2016-01-01             5  2560   2   0   0   
...              ...  ...    ...         ...           ...   ...  ..  ..  ..   
572929  190906210457    6     -1  2021-08-30             1   506   2   0   0   
572930  190906210458    6     -1  2021-08-30             1   752   2   0   0   
572931  190906210459    3   2099  2021-08-30             1  1505   1   3   0   
572932  190906210460    3   2099  2021-08-31             2  1245   0   0   0   
572933  190906210461    3  21029  2021-08-31             2  1834   3   0   4   

        p9  ...      l        n  o                       p       q       r  \
0        2  ...          711403     Souhlasnýsesměremúseku  Pomalý  554782   
1        2  ...                                                         -1   
2        2  ...                                                         -1   
3        2  ...                                                         -1   
4        2  ...                                                         -1   
...     ..  ...    ...      ... ..                     ...     ...     ...   
572929   2  ...         2611387                             Pomalý  560359   
572930   2  ...         2541503         Opačnýkesměruúseku  Pomalý  560383   
572931   2  ...                                                         -1   
572932   1  ...   2099   199764         Opačnýkesměruúseku  Pomalý  560286   
572933   1  ...  21029  3650728     Souhlasnýsesměremúseku  Pomalý  560324   

             s                   t  p5a  region  
0       451622  GN_V0.1UIR-ADR_410    1     PHA  
1           -1                        1     PHA  
2           -1                        1     PHA  
3           -1                        1     PHA  
4           -1                        1     PHA  
...        ...                 ...  ...     ...  
572929      -1  GN_V0.1UIR-ADR_410    1     KVK  
572930  550426  GN_V0.1UIR-ADR_410    1     KVK  
572931      -1                        1     KVK  
572932  553654  GN_V0.1UIR-ADR_410    1     KVK  
572933      -1  GN_V0.1UIR-ADR_410    2     KVK  

[572934 rows x 65 columns]

upravená cena škod z stokorun na koruny (není zapotřebí, pouze pro pohodlí uživatele)

In [10]:
df['p53'] = df['p53'].astype(int).mul(100)

Vytvoříme nové sloupce pro cenu nehod, a jestli účastněné značky aut jsou audi nebo škoda

In [11]:
df['cena'] = df['p53']
df['audi'] = df['p45a'] == 2
df['skoda'] = df['p45a'] == 39

In [12]:
skoda = pd.crosstab(df['cena'], df['skoda'])
audi = pd.crosstab(df['cena'], df['audi'])
skoda

skoda      False  True
cena                  
0         155747  8203
100         1370   240
200          565    74
300          110    13
400            9     7
...          ...   ...
5500000        1     0
6000000        2     0
8050000        1     0
15000000       1     0
20000000       1     0

[317 rows x 2 columns]

In [13]:
audi

audi       False  True
cena                  
0         163186   764
100         1598    12
200          638     1
300          122     1
400           16     0
...          ...   ...
5500000        1     0
6000000        2     0
8050000        1     0
15000000       1     0
20000000       1     0

[317 rows x 2 columns]

použijeme test ttest_ind z knihovny scipy. Podle hodnoty statistic zjistíme jak moc se liší škody nehod mezi značkami škoda a audi

In [14]:
st.stats.ttest_ind(skoda[1],audi[1])

Ttest_indResult(statistic=3.3590464130617304, pvalue=0.0008291475169126504)

## Závěr hypotézy 2
hodnota statistic=3.35904641, z toho plyne že škody nehod na audi byly až 3.5x vyšší než na Škodách.
Potvrzujeme hypotézu